components/employees/add-employee-form/index.tsx

```diff
@@ -376,16 +414,24 @@ const AddEmployee: FC<AddEmployeeProps> = ({ setOpenEmployeeForm }) => {
 
           <FormField
             control={form.control}
-            name={"contractEnd"}
+            name={"contractExpiryDate"}
             render={({ field }) => (
               <FormItem>
                 <FormLabel>{contractExpiryDate?.label}</FormLabel>
 
                 <FormControl>
                   <DatePicker
-                    onSelect={field.onChange}
+                    onSelect={(date) => {
+                      field.onChange(date);
+                      setContractEndDate(date);
+                    }}
                     placeholder={contractExpiryDate?.placeholder}
                     value={field?.value}
+                    disabled={(date: Date) =>
+                      date < new Date("1900-01-01") ||
+                      (hiringDate ? date <= hiringDate : false) ||
+                      (probationEndDate ? date < probationEndDate : false)
+                    }
```

Comment:
this is also used thrice so make it a function and use dates as props

Reply:
@imrankhan-stixor, `Disabled` covers different conditions checked for each data picker. If we create a function, we'd need to add switch/if checks to determine which conditions to run. Thus, there's no harm in handling them separately, as we aren't reusing the same checks elsewhere.

Reply:
you're to make a function like this:

const disableTesting=(data:Date, initialTestDate:Date, secondaryTestDate:Date)=>{
return  date < new Date("1900-01-01") ||
                      (initialTestDate ? date <= initialTestDate : false) ||
                      (secondaryTestDate ? date < secondaryTestDate : false)
}